src/components/ChatWoot.ts

```diff
@@ -7,6 +7,41 @@ export const ChatwootWidget: React.FC = () => {
   const chatwootEnabled = import.meta.env.VITE_FEATURE_CHATWOOT === 'true'
   useEffect(() => {
     if (!chatwootEnabled) return // Add Chatwoot Settings
+    const BASE_URL = getConfig().VITE_CHATWOOT_URL
+    let allowedOrigin = ''
+    try {
+      allowedOrigin = new URL(BASE_URL).origin
+    } catch {
+      allowedOrigin = ''
+    }
+    const chatwootMessageGuard = (event: MessageEvent) => {
+      const dataString = typeof event.data === 'string' ? event.data : undefined
+      if (!dataString || !dataString.startsWith('chatwoot-widget:')) return
+      if (!allowedOrigin || event.origin !== allowedOrigin) {
+        event.stopImmediatePropagation()
+        event.stopPropagation()
+        return
+      }
+      try {
+        const payload = JSON.parse(dataString.replace('chatwoot-widget:', '')) as {
+          event?: string
+          baseUrl?: string
+        }
+        if (payload?.event === 'popupChatWindow' && typeof payload.baseUrl === 'string') {
+          const parsed = new URL(payload.baseUrl, allowedOrigin)
```

Comment:
Might be a stupid question but why do we need allowedOrigin as baseUrl here? I'm struggling finding the reason

Reply:
the whole http check seems a bit redundant/pointless unless I am missing something. feels like the same origin check is sufficient (which we are already using https:// url as the allowed origin as based on our env var)